### ISSUE TITLE
HibernateDomainExporter does not register spatial types

### ISSUE THREAD

**Issue by ridoo** [querydsl/querydsl#2320]
## tl;dr
Is it possible to either

* configure the Maven plugin generating code from a set of `hbm.xml` files
* configure the `JPADomainExporter` to register spatial types 

I use version `4.1.3`. 

## Context

Section `2.1.4. Generating the model from hbm.xml files` describes a way [to generate code from XML mapping files](http://www.querydsl.com/static/querydsl/4.1.3/reference/html_single/#d0e196). Having a simple Spring Boot Application:

```java
@SpringBootApplication
public class Application implements CommandLineRunner {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @Autowired
    @Qualifier("entityManagerFactory")
    private EntityManagerFactory entityManagerFactory;

    @Bean
    @Primary
    public SessionFactory sessionFactory() {
        if (entityManagerFactory.unwrap(SessionFactory.class) == null) {
            throw new NullPointerException("factory is not a hibernate factory");
        }
        return entityManagerFactory.unwrap(SessionFactory.class);
    }

    @Bean
    public EntityManagerFactory entityManagerFactory(DataSource datasource, JpaProperties properties)
            throws IOException {
        LocalContainerEntityManagerFactoryBean emf = new LocalContainerEntityManagerFactoryBean();
        emf.setJpaVendorAdapter(new HibernateJpaVendorAdapter());
        emf.setJpaPropertyMap(properties.getProperties());
        emf.setDataSource(datasource);
        emf.afterPropertiesSet();
        return emf.getNativeEntityManagerFactory();
    }

    @Override
    public void run(String... args) throws Exception {
        createExporter("Q").execute();
    }

    private JPADomainExporter createExporter(String prefix) {
        final Metamodel metamodel = entityManagerFactory.getMetamodel();
        return new JPADomainExporter(prefix, new File("src/main/java"), metamodel);
    }

}
```

I first tried to use `HibernateDomainExporter` but the above config also worked out for `JPADomainExporter`. The mapping files are declared within a `META-INF/persistence.xml` on the classpath, like so:

```xml
<persistence xmlns="http://xmlns.jcp.org/xml/ns/persistence"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence
             http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd"
    version="2.1">
    <persistence-unit name="default-unit">
        <description>Default Mappings</description>
        <mapping-file>/hbm/mapping-resource_1.hbm.xml</mapping-file>
        <mapping-file>/hbm/mapping-resource_2.hbm.xml</mapping-file>

        <!-- et cetera --> 

    </persistence-unit>
</persistence>
```

This works rather well. However, I do have spatial types which are not mapped by default. The documentation hints me to [add `<spatial>true</spatial>` to the Maven plugin config](http://www.querydsl.com/static/querydsl/4.1.3/reference/html_single/#spatial), but I did not find a way to configure the maven codegen plugin to work with `hbm.xml` mappings.

**Comment by ridoo** [comment]
I [opened a PR](https://github.com/querydsl/querydsl/pull/2322) which makes it possible to register spatial paths in a non-Maven run.

**Comment by Shredder121** [comment]
I saw it, thanks, first glance looks good!
Yes if you want it to be used in Querydsl 4, then keeping the old class is needed.

**Comment by ridoo** [comment]
I've restored the old `SpatialSupport` now delegating to `SpatialTypeMapper` which encapsulates the actual type registration. I shamelessly marked `SpatialSupport` as `@Deprecated`, so feel free to revert if you have reasons to not doing so. I am not aware if you have any rules when to deprecate something or not.
